### ISSUE TITLE
Crash: Size "8" not specified in allowed cluster sizes map

### ISSUE THREAD

**Issue by antiguru** [MaterializeInc/materialize#11968]
### What version of Materialize are you using?

85be487e3c77bb828a46141c19f87f4e3017a8f0

### How did you install Materialize?

Built from source

### What was the issue?

Cluster size validation happens in the wrong place:

```
materialize=> drop cluster default cascade;
DROP CLUSTER
materialize=> create cluster default size "8";
server closed the connection unexpectedly
        This probably means the server terminated abnormally
        before or while processing the request.
The connection to the server was lost. Attempting reset: Failed.
```

### Relevant log output

```source
2022-04-21T14:19:47.328900Z ERROR panic: called `Result::unwrap()` on an `Err` value: Size "8" not specified in allowed cluster sizes map
thread: coordinator
location: /home/moritz/dev/repos/materialize/src/coord/src/coord.rs:1889:22
version: 0.26.1-dev (d3b684a1e37f401e18262a144f77130409983ebf)
backtrace:
   0: materialized::handle_panic
             at src/materialized/src/bin/materialized/main.rs:856:21
   1: core::ops::function::Fn::call
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/ops/function.rs:70:5
   2: std::panicking::rust_panic_with_hook
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:702:17
   3: std::panicking::begin_panic_handler::{{closure}}
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:588:13
   4: std::sys_common::backtrace::__rust_end_short_backtrace
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/sys_common/backtrace.rs:138:18
   5: rust_begin_unwind
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:584:5
   6: core::panicking::panic_fmt
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/panicking.rs:143:14
   7: core::result::unwrap_failed
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/result.rs:1749:5
   8: core::result::Result<T,E>::unwrap
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/result.rs:1065:23
      mz_coord::coord::Coordinator::sequence_create_compute_instance::{{closure}}
             at src/coord/src/coord.rs:1882:17
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/future/mod.rs:84:19
      mz_coord::coord::Coordinator::sequence_plan::{{closure}}
             at src/coord/src/coord.rs:1529:68
   9: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/future/mod.rs:84:19
      mz_coord::coord::Coordinator::handle_execute::{{closure}}
             at src/coord/src/coord.rs:1365:66
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/future/mod.rs:84:19
      mz_coord::coord::Coordinator::message_command::{{closure}}
             at src/coord/src/coord.rs:971:62
  10: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/future/mod.rs:84:19
      mz_coord::coord::Coordinator::serve::{{closure}}
             at src/coord/src/coord.rs:688:67
  11: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/future/mod.rs:84:19
      tokio::park::thread::CachedParkThread::block_on::{{closure}}
             at /home/moritz/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/park/thread.rs:263:54
      tokio::coop::with_budget::{{closure}}
             at /home/moritz/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/coop.rs:102:9
      std::thread::local::LocalKey<T>::try_with
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/thread/local.rs:413:16
      std::thread::local::LocalKey<T>::with
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/thread/local.rs:389:9
      tokio::coop::with_budget
             at /home/moritz/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/coop.rs:95:5
      tokio::coop::budget
             at /home/moritz/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/coop.rs:72:5
      tokio::park::thread::CachedParkThread::block_on
             at /home/moritz/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/park/thread.rs:263:31
      tokio::runtime::enter::Enter::block_on
             at /home/moritz/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/runtime/enter.rs:151:13
  12: tokio::runtime::handle::Handle::block_on
             at /home/moritz/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.17.0/src/runtime/handle.rs:355:9
  13: mz_coord::coord::serve::{{closure}}::{{closure}}
             at src/coord/src/coord.rs:4721:17
      std::sys_common::backtrace::__rust_begin_short_backtrace
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/sys_common/backtrace.rs:122:18
  14: std::thread::Builder::spawn_unchecked_::{{closure}}::{{closure}}
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/thread/mod.rs:498:17
      <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/panic/unwind_safe.rs:271:9
      std::panicking::try::do_call
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:492:40
      std::panicking::try
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:456:19
      std::panic::catch_unwind
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panic.rs:137:14
      std::thread::Builder::spawn_unchecked_::{{closure}}
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/thread/mod.rs:497:30
      core::ops::function::FnOnce::call_once{{vtable.shim}}
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/ops/function.rs:227:5
  15: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/alloc/src/boxed.rs:1853:9
      <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/alloc/src/boxed.rs:1853:9
      std::sys::unix::thread::Thread::new::thread_start
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/sys/unix/thread.rs:108:17
  16: start_thread
             at ./nptl/./nptl/pthread_create.c:481:8
  17: clone
             at ./misc/../sysdeps/unix/sysv/linux/x86_64/clone.S:95

materialized encountered an internal error and crashed.

We rely on bug reports to diagnose and fix these errors. Please
copy and paste the above details and file a report at:

    https://materialize.com/s/bug
```
```

**Comment by umanwizard** [comment]
There are two separate issues here:

1. A parsing bug caused us to interpret `"8"` as including the quotes. This is fixed by #11980 .

2. Passing an incorrect size causes us to crash. This is not yet fixed.

**Comment by umanwizard** [comment]
So, after #11980 lands,  `CREATE CLUSTER default SIZE "8"` will no longer crash (it will mean the same thing as the more standard `CREATE CLUSTER default SIZE '8'` , so the exact issue reported here is gone, but I will leave it open because `CREATE CLUSTER default SIZE 'derp'` still causes a crash.
